test/intergration_test.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/vkuznet/dbs2go/dbs"
+	"github.com/vkuznet/dbs2go/web"
+
+	limiter "github.com/ulule/limiter/v3"
+	stdlib "github.com/ulule/limiter/v3/drivers/middleware/stdlib"
+	memory "github.com/ulule/limiter/v3/drivers/store/memory"
+)
+
+func newreq(t *testing.T, method string, url string, body io.Reader) *http.Request {
+	r, err := http.NewRequest(method, url, body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return r
+}
+
+func initTestLimiter(t *testing.T, period string) {
+	rate, err := limiter.NewRateFromFormatted(period)
+	if err != nil {
+		t.Fatalf("Limiter Error")
+	}
+	store := memory.NewStore()
+	instance := limiter.New(store, rate)
+	web.LimiterMiddleware = stdlib.NewMiddleware(instance)
+}
+
+func configTestServer(t *testing.T, serverType string, lexiconFile string) {
+	web.Config.Base = "/dbs"
+	web.Config.DBFile = "../dbfile"
+	web.Config.LexiconFile = lexiconFile
+	web.Config.ServerCrt = ""
+	web.Config.ServerKey = ""
+	web.Config.ServerType = "DBSWriter"
+	web.Config.LogFile = serverType
+	web.Config.Verbose = 0
+
+	initTestLimiter(t, "100-S")
+}
+
+func TestDBSWriterIntegration(t *testing.T) {
+	db := initDB(false)
+	defer db.Close()
+
+	configTestServer(t, "DBSWriter", "../static/lexicon_writer.json")
+
+	ts := httptest.NewServer(web.Handlers())
+	defer ts.Close()
+
+	t.Run("Test POST datatiers", func(t *testing.T) {
```

Comment:
this is fine for first approach but I suggest to generalize function since we'll do plenty of injections and all steps will be identical for different structs. Since all structs in dbs module implement `DBRecord` interface it means that you can create a function which will accept it, e.g.
```
func injectDBRecord(rec dbs.DBRecord) {
}
```
Then, you may call it like
```
    dt := dbs.DataTiers{
			DATA_TIER_NAME: "GEN-SIM-RAW",
			CREATE_BY:      "tester",
		}
    t.Run("Test POST datatiers", innjectDBRecord(dt))
...
// here you can create another struct, e.g. for primary  datasets, etc. and perform t.Run again for it
```